### DISCUSSION TITLE
Cancel transition in q-stepper

### DISCUSSION THREAD

**Discussion by yesennes** [quasarframework/quasar#13225] [Ideas / Proposals]
I have a QStepper which normally allows navigation through the header, by clicking on the different steps. Sometimes, like when the current step has unsaved changes or validation issues, we disallow header navigation and change the color of the other steps.

This is somewhat confusion the the user, as what was clickable now isn't without explanation. I'd like more control over the header navigation to provide a better experience. Some potential solutions:

- Allow the transition to be canceled in before-transition event
- Sends a onClick event when the user clicks on a (disabled) header
- Allow a tootlip over the step headers
- Make the step header a slot to allow custom rendering

Any of those would suffice for our use case, and potentially be helpful for others.

Thanks!

**Comment by rfox12** [comment]
I found that you can pull this off by using a writable computed value (one with a setter).  Then only allow the update if valid....
```
    const innerStep = ref(1);
    const step = computed({
      get() {
        return innerStep.value;
      },
      set(newValue: number) {
        if (innerStep.value !== 1 || validateFields()) { // only move away from step 1 if valid
          innerStep.value = newValue;
        }
      },
    });
```
Then just use `step` as your model for QStepper.
